### ISSUE TITLE
AWSPinpoint makes call to gfx from background thread

### ISSUE THREAD

**Issue by helloniklas** [aws/aws-sdk-ios#734]
### To help us solve your problem better, please answer the following list of questions.

* What service are you using?

AWSPinpoint

* In what version of SDK are you facing the problem?

The latest sdk as of today.

Xcode 9 warns that your SDK is making illegal calls to the gfx from a background thread.

-[UIApplication isRegisteredForRemoteNotifications] must be used from main thread only

This happens in AWSPinpointEndpointProfile.m

```- (BOOL) isSystemLevelOptOut {
    BOOL optOut = ![[UIApplication sharedApplication] isRegisteredForRemoteNotifications];

    if ([[UIApplication sharedApplication] currentUserNotificationSettings].types == UIUserNotificationTypeNone) {
        optOut = YES;
    }

    return optOut;
}```

**Comment by rohandubal** [comment]
Hello @helloniklas 

Thanks for reporting this. We will investigate and revert.

Thanks,
Rohan

**Comment by rohandubal** [comment]
Fixed with 2.6.2.
